### PR TITLE
feat(core): use `TuiBreakpointService` instead of `TUI_IS_MOBILE_RES`

### PR DESCRIPTION
**Pull request by splincode**
## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [x] Refactoring
- [ ] Build or CI related changes
- [ ] Tests related changes
- [ ] Documentation content changes

### CODE REVIEW

File: projects/core/components/root/root.component.ts [hunk 1]
```diff
@@ -28,17 +24,21 @@ import {debounceTime, map} from 'rxjs/operators';
...
     // eslint-disable-next-line @angular-eslint/prefer-on-push-component-change-detection
     changeDetection: ChangeDetectionStrategy.Default,
-    providers: [TUI_IS_MOBILE_RES_PROVIDER],
     encapsulation: ViewEncapsulation.None,
     host: {
         'data-tui-version': TUI_VERSION,
         '[style.--tui-duration.ms]': 'duration',
         '[class._ios]': 'isIOS',
         '[class._android]': 'isAndroid',
+        '[$.class._mobile]': 'isMobileRes$',
```
Comment (waterplea) [inline projects/core/components/root/root.component.ts:33]: I think AOT will fail due to an unknown property.

Comment (splincode) [reply projects/core/components/root/root.component.ts:33]: analogue:
image

Comment (waterplea) [reply projects/core/components/root/root.component.ts:33]: I forgot that it doesn't check `host`

File: projects/core/components/root/root.component.ts [hunk 2]
```diff
@@ -28,17 +24,21 @@ import {debounceTime, map} from 'rxjs/operators';
     // So that we do not force OnPush on custom dialogs
     // eslint-disable-next-line @angular-eslint/prefer-on-push-component-change-detection
     changeDetection: ChangeDetectionStrategy.Default,
-    providers: [TUI_IS_MOBILE_RES_PROVIDER],
```
Comment (waterplea) [inline projects/core/components/root/root.component.ts:31]: Should we delete/deprecate it?

File: projects/core/components/root/root.component.ts [hunk 3]
```diff
@@ -52,7 +52,7 @@ export class TuiRootComponent {
         @Inject(TUI_DIALOGS)
         readonly dialogs: ReadonlyArray<Observable<readonly unknown[]>>,
         @Inject(TUI_IS_MOBILE) private readonly isMobile: boolean,
-        @Inject(TUI_IS_MOBILE_RES) readonly isMobileRes$: Observable<boolean>,
+        @Inject(TuiBreakpointService) readonly breakpoint: TuiBreakpointService,
```
Comment (waterplea) [inline projects/core/components/root/root.component.ts:55]: private

**Review by waterplea — APPROVED** [review]

File: projects/core/components/root/root.component.ts (hunk 3 above)
Comment (splincode) [reply projects/core/components/root/root.component.ts]: fixed

File: projects/core/components/root/root.component.ts (hunk 2 above)
Comment (splincode) [reply projects/core/components/root/root.component.ts:31]: fixed

**Review by vladimirpotekhin — APPROVED** [review]

**Pull request merged**